llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -503,6 +517,21 @@ DemandedFields getDemanded(const MachineInstr &MI, const RISCVSubtarget *ST) {
     }
   }
 
+  // In §32.16.6, whole vector register moves have a dependency on SEW. At the
+  // MIR level though we don't encode the element type, and it gives the same
+  // result whatever the SEW may be.
+  //
+  // However it does need valid SEW, i.e. vill must be cleared. The entry to a
+  // function, calls and inline assembly may all set it, so make sure we clear
+  // it for whole register copies. Do this by leaving VILL demanded.
+  if (isVectorCopy(ST->getRegisterInfo(), MI)) {
```

Comment:
Can you add a cl::opt for testing?  I think simple having it conditional here would effectively disable this change, and having a quick way to rule in/out this change might be useful going forward.

Reply:
I've added it under `-riscv-insert-vsetvli-whole-vector-register-move-valid-vtype=false`. I ended up having to put the check in emitVSETVLIs because it adds an implicit use on VTYPE there, and getting rid of the implicit use seems to undo the change.